src/agent/onefuzz-supervisor/src/commands.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::auth::Secret;
+use anyhow::Result;
+use std::process::Stdio;
+use tokio::{fs, io::AsyncWriteExt, process::Command};
+
+#[cfg(target_os = "windows")]
+use std::{env, path::PathBuf};
+
+#[cfg(target_os = "linux")]
+use users::{get_current_uid, get_user_by_name, get_user_by_uid, os::unix::UserExt};
+
+#[derive(Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct SshKeyInfo {
+    pub key: Secret<String>,
```

Comment:
Maybe rename this `public_key`, though I think keeping it `Secret`-wrapped is fine.